### ISSUE TITLE
Type 'IconDefinition' is not assignable to type 'IconProp' with Typescript

### ISSUE THREAD

**Issue by geoseong** [FortAwesome/react-fontawesome#366]
I'm using typescript and next.js react project.

I was trying to import github icon(`faGithub`) in fontawesome.

when I run the next.js's dev mode first without import github icon (no error happens in typescript) and trying to import github icon in fontawesome, it works well.
```js
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub } from '@fortawesome/free-brands-svg-icons';

...

<FontAwesomeIcon icon={faGithub} className={sizeClass} />
```

but problem is to execute typescript compiler(tsc). tsc throws error like this
```
error TS2322: Type 'IconDefinition' is not assignable to type 'IconProp'.
  Type 'IconDefinition' is not assignable to type 'IconLookup'.
    Types of property 'iconName' are incompatible.
      Type 'import("/[path-to-my-project]/node_modules/@fortawesome/free-brands-svg-icons/node_modules/@fortawesome/fontawesome-common-types/index").IconName' is not assignable to type 'import("/[path-to-my-project]/node_modules/@fortawesome/fontawesome-common-types/index").IconName'.
        Type '"deezer"' is not assignable to type 'IconName'.

26         <FontAwesomeIcon icon={faGithub} className={sizeClass} />
```

I'd tried using some ways, `tsc` passed. but there's not working in actual react project
- <FontAwesomeIcon icon={['fab', 'github']} className={sizeClass} />
- <FontAwesomeIcon icon={['fas', 'github']} className={sizeClass} />
```
Could not find icon {prefix: "fab", iconName: "github"} 

Could not find icon {prefix: "fas", iconName: "github"} 
```

so I tried this way to make it work the `tsc` and actual react project.
```js
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { faGithub } from '@fortawesome/free-brands-svg-icons';

...

const faPropIcon = faGithub as IconProp;

...

<FontAwesomeIcon icon={faPropIcon} className={sizeClass} />
```

my package.json
```
"@fortawesome/fontawesome-svg-core": "^1.2.28",
"@fortawesome/free-brands-svg-icons": "^5.14.0",
"@fortawesome/free-solid-svg-icons": "^5.13.0",
"@fortawesome/react-fontawesome": "^0.1.9",
```

I think it's kind of weird.

because this problem happens after installing `@fortawesome/free-brands-svg-icons` and use `faGithub` in `@fortawesome/free-brands-svg-icons`.

is it bug? or should I use this way like this continuously?

**Comment by toledoal** [comment]
I'm having the same issue... It used to work well. Maybe a change in typescript?

**Comment by eseceve** [comment]
I'm having the same issue but only for regular icons.

```jsx
import React from 'react';
import { faMapMarkerAlt as regularIcon } from '@fortawesome/pro-regular-svg-icons/faMapMarkerAlt';
import { faMapMarkerAlt as solidIcon } from '@fortawesome/pro-solid-svg-icons/faMapMarkerAlt';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export const myRegularIcon = () =>  (<FontAwesomeIcon icon={regularIcon} />); // fails
export const mySolidIcon = () =>  (<FontAwesomeIcon icon={solidIcon} />); // works
```

**Comment by chloe-codes1** [comment]
Having the same issue as well.

**Comment by MaximeKoitsalu** [comment]
Also having the same issue 🤔

**Comment by robmadole** [comment]
Can you all try something for me (these Typescript issues are so frustrating for everyone)?

Upgrade to the latest dependencies on everything (including `@fortawesome/fontawesome-common-types`) and give this another go. And if anyone can provide a reproducible test case that would be very helpful. Every time I try to reproduce a TS error I fail so I must be missing a critical piece. We don't use TS daily at Font Awesome so we have to rely pretty heavily on the community to solve some of these.

**Comment by keishibuki** [comment]
A similar error occurred in the following versions,
but updating the package dependencies makes the error go away.

### Before
```
"@fortawesome/fontawesome-common-types@^0.2.27":
  version "0.2.27"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-common-types/-/fontawesome-common-types-0.2.27.tgz#19706345859fc46adf3684ed01d11b40903b87e9"
  integrity sha512-97GaByGaXDGMkzcJX7VmR/jRJd8h1mfhtA7RsxDBN61GnWE/PPCZhOdwG/8OZYktiRUF0CvFOr+VgRkJrt6TWg==

"@fortawesome/fontawesome-free@^5.12.1":
  version "5.12.1"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-free/-/fontawesome-free-5.12.1.tgz#2a98fea9fbb8a606ddc79a4680034e9d5591c550"
  integrity sha512-ZtjIIFplxncqxvogq148C3hBLQE+W3iJ8E4UvJ09zIJUgzwLcROsWwFDErVSXY2Plzao5J9KUYNHKHMEUYDMKw==

"@fortawesome/fontawesome-svg-core@^1.2.27":
  version "1.2.27"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-svg-core/-/fontawesome-svg-core-1.2.27.tgz#e4db8e3be81a40988213507c3e3d0c158a6641a3"
  integrity sha512-sOD3DKynocnHYpuw2sLPnTunDj7rLk91LYhi2axUYwuGe9cPCw7Bsu9EWtVdNJP+IYgTCZIbyARKXuy5K/nv+Q==
  dependencies:
    "@fortawesome/fontawesome-common-types" "^0.2.27"

"@fortawesome/free-brands-svg-icons@^5.12.1":
  version "5.12.1"
  resolved "https://registry.yarnpkg.com/@fortawesome/free-brands-svg-icons/-/free-brands-svg-icons-5.12.1.tgz#67977addd15e21e57aa1ed71cd2ddecdfaa88647"
  integrity sha512-IYUYcgGsQuwiIHjRGfeSTCIQKUSZMb6FsV6mDj78K0D+YzGJkM4cvEBBUMHtnla5D2HCxncMI/9JX5YIk2GHeQ==
  dependencies:
    "@fortawesome/fontawesome-common-types" "^0.2.27"

"@fortawesome/react-fontawesome@^0.1.9":
  version "0.1.9"
  resolved "https://registry.yarnpkg.com/@fortawesome/react-fontawesome/-/react-fontawesome-0.1.9.tgz#c865b9286c707407effcec99958043711367cd02"
  integrity sha512-49V3WNysLZU5fZ3sqSuys4nGRytsrxJktbv3vuaXkEoxv22C6T7TEG0TW6+nqVjMnkfCQd5xOnmJoZHMF78tOw==
  dependencies:
    prop-types "^15.7.2"
```

### After
```
"@fortawesome/fontawesome-common-types@^0.2.32":
  version "0.2.32"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-common-types/-/fontawesome-common-types-0.2.32.tgz#3436795d5684f22742989bfa08f46f50f516f259"
  integrity sha512-ux2EDjKMpcdHBVLi/eWZynnPxs0BtFVXJkgHIxXRl+9ZFaHPvYamAfCzeeQFqHRjuJtX90wVnMRaMQAAlctz3w==

"@fortawesome/fontawesome-free@^5.12.1":
  version "5.15.1"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-free/-/fontawesome-free-5.15.1.tgz#ccfef6ddbe59f8fe8f694783e1d3eb88902dc5eb"
  integrity sha512-OEdH7SyC1suTdhBGW91/zBfR6qaIhThbcN8PUXtXilY4GYnSBbVqOntdHbC1vXwsDnX0Qix2m2+DSU1J51ybOQ==

"@fortawesome/fontawesome-svg-core@^1.2.27":
  version "1.2.32"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-svg-core/-/fontawesome-svg-core-1.2.32.tgz#da092bfc7266aa274be8604de610d7115f9ba6cf"
  integrity sha512-XjqyeLCsR/c/usUpdWcOdVtWFVjPbDFBTQkn2fQRrWhhUoxriQohO2RWDxLyUM8XpD+Zzg5xwJ8gqTYGDLeGaQ==
  dependencies:
    "@fortawesome/fontawesome-common-types" "^0.2.32"

"@fortawesome/free-brands-svg-icons@^5.12.1":
  version "5.15.1"
  resolved "https://registry.yarnpkg.com/@fortawesome/free-brands-svg-icons/-/free-brands-svg-icons-5.15.1.tgz#1dc0563f4036639e53d24b8e532ea78a53ca2250"
  integrity sha512-pkTZIWn7iuliCCgV+huDfZmZb2UjslalXGDA2PcqOVUYJmYL11y6ooFiMJkJvUZu+xgAc1gZgQe+Px12mZF0CA==
  dependencies:
    "@fortawesome/fontawesome-common-types" "^0.2.32"

"@fortawesome/react-fontawesome@^0.1.9":
  version "0.1.11"
  resolved "https://registry.yarnpkg.com/@fortawesome/react-fontawesome/-/react-fontawesome-0.1.11.tgz#c1a95a2bdb6a18fa97b355a563832e248bf6ef4a"
  integrity sha512-sClfojasRifQKI0OPqTy8Ln8iIhnxR/Pv/hukBhWnBz9kQRmqi6JSH3nghlhAY7SUeIIM7B5/D2G8WjX0iepVg==
  dependencies:
    prop-types "^15.7.2"
```

**Comment by geoseong** [comment]
Let me update

I upgrade latest version, it works well. error is gone without using `as`.

```js
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
- import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { faGithub } from '@fortawesome/free-brands-svg-icons';

...

- const faPropIcon = faGithub as IconProp;
...

- <FontAwesomeIcon icon={faPropIcon} className={sizeClass} />
+ <FontAwesomeIcon icon={faGithub} className={sizeClass} />
```

## package.json
```
"@fortawesome/fontawesome-svg-core": "^1.2.28", -> "^1.2.32",
"@fortawesome/free-brands-svg-icons": "^5.14.0", -> "^5.15.1",
"@fortawesome/free-solid-svg-icons": "^5.13.0", -> "^5.15.1",
```

## yarn.lock
### Before
```
"@fortawesome/fontawesome-common-types@^0.2.28":
  version "0.2.28"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-common-types/-/fontawesome-common-types-0.2.28.tgz#1091bdfe63b3f139441e9cba27aa022bff97d8b2"
  integrity sha512-gtis2/5yLdfI6n0ia0jH7NJs5i/Z/8M/ZbQL6jXQhCthEOe5Cr5NcQPhgTvFxNOtURE03/ZqUcEskdn2M+QaBg==

"@fortawesome/fontawesome-common-types@^0.2.30":
  version "0.2.30"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-common-types/-/fontawesome-common-types-0.2.30.tgz#2f1cc5b46bd76723be41d0013a8450c9ba92b777"
  integrity sha512-TsRwpTuKwFNiPhk1UfKgw7zNPeV5RhNp2Uw3pws+9gDAkPGKrtjR1y2lI3SYn7+YzyfuNknflpBA1LRKjt7hMg==

"@fortawesome/fontawesome-svg-core@^1.2.28":
  version "1.2.28"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-svg-core/-/fontawesome-svg-core-1.2.28.tgz#e5b8c8814ef375f01f5d7c132d3c3a2f83a3abf9"
  integrity sha512-4LeaNHWvrneoU0i8b5RTOJHKx7E+y7jYejplR7uSVB34+mp3Veg7cbKk7NBCLiI4TyoWS1wh9ZdoyLJR8wSAdg==
  dependencies:
    "@fortawesome/fontawesome-common-types" "^0.2.28"

"@fortawesome/free-brands-svg-icons@^5.14.0":
  version "5.14.0"
  resolved "https://registry.yarnpkg.com/@fortawesome/free-brands-svg-icons/-/free-brands-svg-icons-5.14.0.tgz#98555518ba41bdff82fbae2f4d1bc36cd3b1c043"
  integrity sha512-WsqPFTvJFI7MYkcy0jeFE2zY+blC4OrnB9MJOcn1NxRXT/sSfEEhrI7CwzIkiYajLiVDBKWeErYOvpsMeodmCQ==
  dependencies:
    "@fortawesome/fontawesome-common-types" "^0.2.30"

"@fortawesome/free-solid-svg-icons@^5.13.0":
  version "5.13.0"
  resolved "https://registry.yarnpkg.com/@fortawesome/free-solid-svg-icons/-/free-solid-svg-icons-5.13.0.tgz#44d9118668ad96b4fd5c9434a43efc5903525739"
  integrity sha512-IHUgDJdomv6YtG4p3zl1B5wWf9ffinHIvebqQOmV3U+3SLw4fC+LUCCgwfETkbTtjy5/Qws2VoVf6z/ETQpFpg==
  dependencies:
    "@fortawesome/fontawesome-common-types" "^0.2.28"
```

### After
```
"@fortawesome/fontawesome-common-types@^0.2.32":
  version "0.2.32"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-common-types/-/fontawesome-common-types-0.2.32.tgz#3436795d5684f22742989bfa08f46f50f516f259"
  integrity sha512-ux2EDjKMpcdHBVLi/eWZynnPxs0BtFVXJkgHIxXRl+9ZFaHPvYamAfCzeeQFqHRjuJtX90wVnMRaMQAAlctz3w==

"@fortawesome/fontawesome-svg-core@^1.2.32":
  version "1.2.32"
  resolved "https://registry.yarnpkg.com/@fortawesome/fontawesome-svg-core/-/fontawesome-svg-core-1.2.32.tgz#da092bfc7266aa274be8604de610d7115f9ba6cf"
  integrity sha512-XjqyeLCsR/c/usUpdWcOdVtWFVjPbDFBTQkn2fQRrWhhUoxriQohO2RWDxLyUM8XpD+Zzg5xwJ8gqTYGDLeGaQ==
  dependencies:
    "@fortawesome/fontawesome-common-types" "^0.2.32"

"@fortawesome/free-brands-svg-icons@^5.15.1":
  version "5.15.1"
  resolved "https://registry.yarnpkg.com/@fortawesome/free-brands-svg-icons/-/free-brands-svg-icons-5.15.1.tgz#1dc0563f4036639e53d24b8e532ea78a53ca2250"
  integrity sha512-pkTZIWn7iuliCCgV+huDfZmZb2UjslalXGDA2PcqOVUYJmYL11y6ooFiMJkJvUZu+xgAc1gZgQe+Px12mZF0CA==
  dependencies:
    "@fortawesome/fontawesome-common-types" "^0.2.32"

"@fortawesome/free-solid-svg-icons@^5.15.1":
  version "5.15.1"
  resolved "https://registry.yarnpkg.com/@fortawesome/free-solid-svg-icons/-/free-solid-svg-icons-5.15.1.tgz#e1432676ddd43108b41197fee9f86d910ad458ef"
  integrity sha512-EFMuKtzRMNbvjab/SvJBaOOpaqJfdSap/Nl6hst7CgrJxwfORR1drdTV6q1Ib/JVzq4xObdTDcT6sqTaXMqfdg==
  dependencies:
    "@fortawesome/fontawesome-common-types" "^0.2.32"
```

Let me close this issue?

**Comment by sishanov** [comment]
import { faStar } from "@fortawesome/free-solid-svg-icons";
. . .
return `(<FontAwesomeIcon` icon={faStar} fontSize="50" />)
. . .

This worked for me.
Seems like the icon is not a string and it is of type Iconprop so you need to assign a proper object to it that can be imported from "@fortawesome/free-solid-svg-icons"; and easily use object destructive to get a specific Icon

**Comment by himadhar** [comment]
I know this is closed, but if it helps, I did this and it worked. :)

`<FontAwesomeIcon icon={faFacebookF as IconProp}`

**Comment by Speedy1991** [comment]
@Jule- I can confirm this
Upgrading all packages to the newest version fixed this for me
(especially `@fortawesome/fontawesome-svg-core": "^1.3.0` -> `@fortawesome/fontawesome-svg-core": "^6.1.0`)

**Comment by karlkaspar** [comment]
Type '"fa-solid fa-arrow-right-from-bracket"' is not assignable to type 'IconProp'.

<FontAwesomeIcon icon="fa-solid fa-arrow-right-from-bracket" />

Still the same problem

**Comment by karlkaspar** [comment]
I have come to understand the underlying issue.

I followed the instructions found on Fontawesome website to install their product.

Plain incompetence at Fontawesome dev team... and frankly I do not think anybody at Fontawesome even cares, because this is such a dumb mistake.
![image](https://user-images.githubusercontent.com/27421479/163716120-7d6289f0-e76a-4e42-b319-c173c6124081.png)
From the image we can see how we are supposed to use an icon in react.
This will give you the error if you are using Typescript, because god forbid anyone and Fontawesome has ever heard of React and Typescript, the most popular web development combo for the past 5 years.

This is how you can actually use it.
<FontAwesomeIcon icon={['fa', 'arrow-right-from-bracket']} />

The first parameter 'fa' can be of these types
export type IconPrefix = "fas" | "far" | "fal" | "fat" | "fad" | "fab" | "fak" ;

THERE IS NO 'FA' IN THE ALLOWED TYPES... the icon Fontawesome said on their page, that I can use, I CANNOT because of their multiple blunders..

The last parameter can be found in  the same file as IconPrefix
export type IconName = [ ... . .. .... . ] tons of lines of code here I am not going to paste.

**Comment by Speedy1991** [comment]
> Plain incompetence at Fontawesome dev team... and frankly I do not think anybody at Fontawesome even cares, because this is such a dumb mistake.


@karlkaspar I think you should rethink your words, they are really inappropriate.

This is an open and friendly community, there is no need to flame/blame others.

Why not just open a new issue (maybe even a PR) to fix the docs? You can spread your knowlege instead of going toxic.

**Comment by SeanBarker182** [comment]
I'm using Font Awesome on a React Native project and I followed [these docs to set it up](https://www.npmjs.com/package/@fortawesome/react-native-fontawesome).

I was getting this issue until I noticed that I missed this install:

`npm i --save @fortawesome/fontawesome-svg-core`

Adding that package resolved my issue.

**Comment by vipulDessai** [comment]
just import any icon from solid svg module
```tsx
import { faPlus } from '@fortawesome/free-solid-svg-icons';

// inside JSX
<FontAwesomeIcon icon={faPlus}/>
```

**Comment by Zeragamba** [comment]
Had the same issue today. Fixed by upgrading packages:
```json
"dependencies": {
    "@fortawesome/fontawesome-svg-core": "^6.1.1",
    "@fortawesome/free-solid-svg-icons": "^6.1.1",
    "@fortawesome/react-fontawesome": "^0.1.18"
}
```

**Comment by iway1** [comment]
> there is no need to flame/blame others.

The devs of this library released a breaking change in a minor version. If anything in the world deserves flaming that does. 

I just had to pull down a project that someone else had to created that uses this library, and I can't even work on it because it doesn't run because there's a typescript compilation error introduced by the new types from a new minor release. Our client is paying us a lot of money to add features to his web app, and now his money is going down the drain because of this mistake and lack of competence of the developers of this library. 

It's fun to be nice and friendly all the time, but people are negatively affected in a very real way by this stuff. The team here either doesn't care enough to do a good job or just is incompetent. There's no other explanation. IMO it helps the community more to post negative stuff here because hopefully we can convince people not to use this library, considering the devs have proven they don't know how to maintain it.

**Comment by richardantao** [comment]
```
{
  "@fortawesome/fontawesome-pro": "^6.1.2",
  "@fortawesome/fontawesome-pro-light": "^5.1.0-3",
  "@fortawesome/fontawesome-pro-solid": "^5.1.0-3",
  "@fortawesome/fontawesome-svg-core": "^6.1.2",
  "@fortawesome/free-brands-svg-icons": "^6.1.2",
  "@fortawesome/pro-duotone-svg-icons": "^6.1.2",
  "@fortawesome/pro-light-svg-icons": "^6.1.2",
  "@fortawesome/pro-regular-svg-icons": "^6.1.2",
  "@fortawesome/pro-solid-svg-icons": "^6.1.2",
  "@fortawesome/pro-thin-svg-icons": "^6.1.2",
  "@fortawesome/react-fontawesome": "^0.1.18"
}
```

The versions that worked for me. Bumping `@fortawesome/react-fontawesome` to `~0.2.x` seems to break the types.

**Comment by JLHasson** [comment]
With the following version & following the steps [here for Next.js](https://fontawesome.com/docs/web/use-with/react/use-with#using-the-icons-in-pages) I don't get any type errors.

```
    "@fortawesome/fontawesome-common-types": "^6.1.2",
    "@fortawesome/fontawesome-svg-core": "^6.1.2",
    "@fortawesome/free-brands-svg-icons": "^6.1.2",
    "@fortawesome/free-regular-svg-icons": "^6.1.2",
    "@fortawesome/free-solid-svg-icons": "^6.1.2",
    "@fortawesome/react-fontawesome": "^0.2.0",
```

**Comment by mauriciosoares** [comment]
If you're having this issue with version `0.2.0`, upgrading the svg icon packages to `6.1.2` fixed the issue for me.

**Comment by mseodev** [comment]
I've tried almost all the combinations posted here already, no luck.
Here's the latest set of versions that's breaking for me. Any ideas?

    "@fortawesome/fontawesome-common-types": "^6.2.0",
    "@fortawesome/fontawesome-svg-core": "^6.2.0",
    "@fortawesome/free-solid-svg-icons": "^6.2.0",
    "@fortawesome/react-fontawesome": "^0.2.0",

**Comment by robmadole** [comment]
@mseodev can you run `npm list | grep fortawesome`?

**Comment by pdanes-rac** [comment]
> @mseodev can you run `npm list | grep fortawesome`?

@robmadole Encountering same issue with same dependency list as @mseodev.

```
> npm list | wsl grep fortawesome
????????? @fortawesome/fontawesome-svg-core@6.2.0
????????? @fortawesome/free-brands-svg-icons@6.2.0
????????? @fortawesome/free-solid-svg-icons@6.2.0
????????? @fortawesome/react-fontawesome@0.2.0
```

**Comment by timgavin** [comment]
I am also having this problem on a brand New Ionic install.

Followed the [Readme instructions](https://github.com/FortAwesome/angular-fontawesome) on this repo

```
$npm list | grep fortawesome

@fortawesome/angular-fontawesome@0.11.1
@fortawesome/fontawesome-svg-core@6.2.0
@fortawesome/pro-light-svg-icons@5.15.4
@fortawesome/pro-regular-svg-icons@5.15.4
@fortawesome/pro-solid-svg-icons@5.15.4

@anglar/cli 14.2.1
```

**Comment by georgiantonov** [comment]
> This is how you can actually use it.
><FontAwesomeIcon icon={['fa', 'arrow-right-from-bracket']} />
>
>The first parameter 'fa' can be of these types
>export type IconPrefix = "fas" | "far" | "fal" | "fat" | "fad" | "fab" | "fak" ;

"Toxic" or no, this solved my issue. No one likes to waste hours because of confusing and wrong documentation.

**Comment by GeorgEchterling** [comment]
This is a dependency issue. Upgrading `@fortawesome/free-solid-svg-icons` from `^6.2.0` to `^6.2.1` causes its dependency on `@fortawesome/fontawesome-common-types` 6.2.1 to be installed separately at `node_modules/@fortawesome/free-solid-svg-icons/node_modules/@fortawesome/fontawesome-common-types` while the existing package `@fortawesome/fontawesome-common-types` installed at `node_modules/@fortawesome/fontawesome-common-types` stays at version 6.2.0. Since `@fortawesome/react-fontawesome` and `@fortawesome/free-solid-svg-icons` use different versions of the same dependency, their types are incompatible. `npm update` fixes it, but the error message is not obvious.

**Comment by LucasHBG** [comment]
As mentioned above by @GeorgEchterling, you should update your packages and set them to the same version.

To verify their versions, type: `yarn list | grep fontawesome` or `npm list | grep fortawesome`.
This command will return something like this:
```
yarn list | grep fontawesome
├─ @fortawesome/fontawesome-common-types@6.2.0
├─ @fortawesome/fontawesome-svg-core@6.2.0
│  └─ @fortawesome/fontawesome-common-types@6.2.0
│  └─ @fortawesome/fontawesome-common-types@6.2.0
│  ├─ @fortawesome/fontawesome-common-types@6.2.1
│  └─ @fortawesome/fontawesome-common-types@6.2.1
├─ @fortawesome/react-fontawesome@0.2.0
```

Then type: `npm update` or `yarn update` (or update one by one). After updating, it should work as expected.
Output:
```
yarn list | grep fontawesome
├─ @fortawesome/fontawesome-common-types@6.2.1
├─ @fortawesome/fontawesome-svg-core@6.2.1
│  └─ @fortawesome/fontawesome-common-types@6.2.1
│  └─ @fortawesome/fontawesome-common-types@6.2.1
│  └─ @fortawesome/fontawesome-common-types@6.2.1
├─ @fortawesome/react-fontawesome@0.2.0
```

**Comment by cjmaret** [comment]
@geoseong 's use of "free-brands-svg-icons" helped me. I made the change below (while guessing that `"fa-solid fa-caret-down"` could be changed to `{ faCaretDown}`. I also had installed free-solid instead of free-brands, but it still worked for me.

`
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+ import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
- <FontAwesomeIcon icon='fa-solid fa-caret-down' />
+ <FontAwesomeIcon icon={faCaretDown}/>

**Comment by brunogroth** [comment]
> @geoseong 's use of "free-brands-svg-icons" helped me. I made the change below (while guessing that `"fa-solid fa-caret-down"` could be changed to `{ faCaretDown}`. I also had installed free-solid instead of free-brands, but it still worked for me.
> 
> ```
> import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
> + import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
> - <FontAwesomeIcon icon='fa-solid fa-caret-down' />
> + <FontAwesomeIcon icon={faCaretDown}/>
> ```

Thanks a lot cjmaret! That was the problem on my project.

**Comment by yiyi1026** [comment]
This works for me.
import { fas } from '@fortawesome/free-solid-svg-icons'
import { fab } from '@fortawesome/free-brands-svg-icons'
import { far } from '@fortawesome/free-regular-svg-icons'
import { library, IconPack } from '@fortawesome/fontawesome-svg-core'

library.add(
far as IconPack,
fas as IconPack,
fab as IconPack
)
`
from https://github.com/FortAwesome/angular-fontawesome/issues/125#issuecomment-1206780101

**Comment by RicFer01** [comment]
I upgraded some packages today `free-solid-svg-icons` and I had this issue, I had to downgrade to the previous release in order to fix it, I think this issue should be re-opened again.
```
 "@fortawesome/fontawesome-svg-core": "^6.2.1",
 "@fortawesome/free-brands-svg-icons": "^6.2.1",
 "@fortawesome/free-solid-svg-icons": "^6.3.0",
 "@fortawesome/react-fontawesome": "^0.2.0",
```

```
 "@fortawesome/fontawesome-svg-core": "^6.2.1",
 "@fortawesome/free-brands-svg-icons": "^6.2.1",
 "@fortawesome/free-solid-svg-icons": "^6.2.1",
 "@fortawesome/react-fontawesome": "^0.2.0",
```

**Comment by Jule-** [comment]
I think this issue should be locked since many broken fixes are shared here.

1. Maintainers should document this more clearly, and potentially setup peer dependencies with more restrictions to reflect hidden version requirements.
2. (As being said many times in this thread) People reading here and trying to upgrade should be aware that **you have to keep all @fortawesome dependencies versions in sync**, do not update one dependency while keeping another back (double-check your lock files with `npm ls` or equivalent to find version mismatch)

**Comment by RicFer01** [comment]
By the way, upgrading also `@fortawesome/free-brands-svg-icons` sees to work
```
"@fortawesome/free-brands-svg-icons": "^6.3.0",
"@fortawesome/free-solid-svg-icons": "^6.3.0",
```

**Comment by biladinn** [comment]
Fellas I didn't have time to read everything on this issue but I managed to make it work in my project. My project is nextjs&typescript one. I followed instructions on Font Awsome website for installing these icons on react project. But it didn't work. So what I did is that I installed --save @fortawesome/free-brands-svg-icons in addition to all of others that they instruct on their website. Btw I needed github icon. And after that I imported into my page this:

`import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub as faGitHub } from '@fortawesome/free-brands-svg-icons';`

This is component that I pasted at place where I want this icon:

`<FontAwesomeIcon icon={faGitHub} bounce />`

I do now know if this will help anyone else but I will leave it here since it helped me.

**Comment by hopingsteam** [comment]
I upgraded to the latest version and the error still appears

**Comment by Illutax** [comment]
> I upgraded to the latest version and the error still appears

`npm outdated` or `npm update` does not take into account the pro dependencies, they have to be updated manually in the package.json to the **same** version.

**Comment by zacherkkila** [comment]
Not sure I saw this explicitly listed as an answer.. For me the core was not the same version as the others.. If this is indeed important, it would be great to have a large note added to the instructions to the website (if it isn't already). I assume many people start with the free version and then when they add pro the versions don't line up.. No guarantee that another yarn install didn't just fix it and the versions were irrelevant, just giving another datapoint.

```
"@fortawesome/fontawesome-svg-core": "^6.4.0",
"@fortawesome/free-solid-svg-icons": "^6.4.2",
"@fortawesome/pro-solid-svg-icons": "^6.4.2",
```

Changing them all at the same version fixed this type error.

```
"@fortawesome/fontawesome-svg-core": "^6.4.2",
"@fortawesome/free-solid-svg-icons": "^6.4.2",
"@fortawesome/pro-solid-svg-icons": "^6.4.2",
```

**Comment by Illutax** [comment]
> Updating all dependencies, also the pro to the same version also helps prevent these errors.

@zacherkkila that is also what I suggested 5 above you. :)

**Comment by pokrovskyy** [comment]
**TL;DR** - the issue seems to be introduced by adding different `@fortawesome/*` package versions and thus is resolved by properly aligning versioning in `package.json`

**Comment by EduardoRamosB** [comment]
Two ways to use it globally:
```
import { library } from '@fortawesome/fontawesome-svg-core'
ibrary.add(faComment)
<FontAwesomeIcon icon={faComment} /> // Inside your component
```
```
import { library, IconProp } from '@fortawesome/fontawesome-svg-core'
<FontAwesomeIcon icon={fa-solid fa-user-astronaut as IconProp} /> // In case you copy/paste directly from FontAwesome
```

**Comment by chuck-waters** [comment]
> I'm using Font Awesome on a React Native project and I followed [these docs to set it up](https://www.npmjs.com/package/@fortawesome/react-native-fontawesome).
> 
> I was getting this issue until I noticed that I missed this install:
> 
> `npm i --save @fortawesome/fontawesome-svg-core`
> 
> Adding that package resolved my issue.

This worked for me too, thanks @SeanBarker182

**Comment by andreifilip123** [comment]
Hello. Me and my team are on the latest version of each package:
```
    "@fortawesome/fontawesome-svg-core": "6.5.2",
    "@fortawesome/free-regular-svg-icons": "6.5.2",
    "@fortawesome/free-solid-svg-icons": "6.5.2",
    "@fortawesome/react-fontawesome": "0.2.0",
```
We even fixed the versions to remove the risk of them being on different versions and we still get this error. We're on typescript 5.4.3. Any clue on how to fix it ?

**Comment by robmadole** [comment]
@andreifilip123 try pinning `@fortawesome/fontawesome-common-types` to `6.5.2` as well.

**Comment by bbehling** [comment]
I had this same problem after installing @fortawesome/free-regular-svg-icons to a latest version.

@fortawesome/free-solid-svg-icons, @fortawesome/fontawesome-svg-core, where on different minor versions. After  updating all three to the same version, error went away.

**Comment by leslieccy** [comment]
FYI, The same issue happens again when updating from `"@fortawesome/free-brands-svg-icons": "^6.5.2"` to `"@fortawesome/free-brands-svg-icons": "^6.6.0"`.

**Comment by chuck-waters** [comment]
I just typecast it and it worked for me.

`... icon={faUserCircle as IconProp} ...`

**Comment by wavded** [comment]
Also same issue as @hugosvirak , I reverted back to 6.6.0 to fix.

**Comment by robmadole** [comment]
Make sure that the `@fortawesome/fontawesome-common-types` gets upgrade to 6.7.1 as well. This is normally the issue when we see these compatibility things upon release.

**Comment by kmsimpel** [comment]
Same issue: I didn't have the fontawesome-common-types dependency before, I installed it

<img width="442" alt="image" src="https://github.com/user-attachments/assets/e9d51b5c-ff6a-4ccd-b9b7-d66dc2776b69" />
